blockchain/hedera.go

```diff
@@ -147,6 +234,79 @@ func (c Client) GetAccountCreditTransactionsAfterTimestamp(accountID string, fro
 	return c.getTransactionsByQuery(transactionsDownloadQuery)
 }
 
+func (c Client) GetAccountByAccountId(accountID string) (*Response, error) {
+	accountQuery := fmt.Sprintf("?account.id=%s", accountID)
+	return c.getAccountByQuery(accountQuery)
+}
+
+// WaitForTransaction Polls the transaction at intervals.
+func (c Client) WaitForTransaction(accoutId string) {
+
+	logger.Infof("Listening for events on account id: %v", accoutId)
+	lastTransactionTimestamp := time.Now().UnixNano()
+	for {
+		response, err := c.GetAccountCreditTransactionsAfterTimestamp(accoutId, lastTransactionTimestamp)
+
+		if err != nil {
+			logger.Errorf("Error while trying to get account. Error: [%s].", err.Error())
+			return
+		}
+
+		if response != nil {
+			numberOfTransactions := len(response.Transactions)
+			if numberOfTransactions != 0 {
+				transactionTimestamp, err := FromString(response.Transactions[numberOfTransactions-1].ConsensusTimestamp)
+				if err != nil {
+					logger.Errorf(err.Error())
+					return
+				}
+				lastTransactionTimestamp = transactionTimestamp
+			}
+			for _, transaction := range response.Transactions {
+				if !checkForValidTokenTransfer(transaction.TokenTransfers, accoutId) {
+					continue
+				}
+
+				if len(transaction.MemoBase64) == 0 {
+					continue
+				}
+
+				decodedMemo, err := DecodeMemo(transaction.MemoBase64)
+				if err != nil {
+					logger.Error("Failed decoding base64 NEAROracleRequestArgs.RequestSpec:", err)
+					continue
+				}
+
+				hederaTransactionInfo := strings.Fields(decodedMemo)
+				if len(hederaTransactionInfo) != 2 {
+					logger.Error("Invalid transaction info format")
+					continue
+				}
+				extractedTopicId := hederaTransactionInfo[0]
+				extractedJobId := hederaTransactionInfo[1]
+
+				if !containsJobId(hederaSubscribersMap[accoutId], extractedJobId) {
+					continue
+				} else {
+					hederaSubs := hederaSubscribersMap[accoutId]
+					for _, hs := range hederaSubs {
+						if hs.jobid == extractedJobId {
+							eventInfo := fmt.Sprintf("{\"hederaTopicId\": \"%s\"}", extractedTopicId)
```

Comment:
Can we switch this to a struct, which will later be marshalled?

Reply:
done